config/http_config.go

```diff
@@ -670,6 +683,14 @@ func NewRoundTripperFromConfigWithContext(ctx context.Context, cfg HTTPClientCon
 		return rt, nil
 	}
 
+	if opts.newTLSConfigFunc != nil {
+		tlsConfig, err := opts.newTLSConfigFunc()
+		if err != nil {
+			return nil, err
+		}
+		return newRT(tlsConfig)
```

Comment:
the first version wasn't short circuiting here, any reason why not to follow branches below?

Reply:
It's because we don't expect custom tlsConfig to be runtime reloadable as it might be built outside of the provided HTTP config. I can keep the old pattern if we want.

Reply:
Ok, I did exact 1:1: NewTLSConfigWithContext semantics so it's simpler on our side.